dbms/src/Storages/Distributed/DirectoryMonitor.cpp

```diff
@@ -167,7 +167,7 @@ ConnectionPoolPtr StorageDistributedDirectoryMonitor::createPool(const std::stri
         return std::make_shared<ConnectionPool>(
             1, host, port, default_database,
             user, password, timeouts,
-            storage.getName() + '_' + name,
+            storage.getName() + '_' + user,
```

Comment:
Why we still have `name` in lambda capture?
What is the purpose of `name` function argument?

Reply:
first - fixed.
second - used in next line
175     auto pools = createPoolsForAddresses(name, pool_factory);